presto-main/src/main/java/com/facebook/presto/execution/DedupingWarningCollector.java

```diff
@@ -13,21 +13,33 @@
  */
 package com.facebook.presto.execution;
 
+import com.facebook.presto.spi.QueryId;
 import com.facebook.presto.spi.WarningCode;
 import com.google.common.collect.ImmutableList;
 
 import java.util.LinkedHashSet;
 import java.util.List;
 import java.util.Set;
 
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+
 public class DedupingWarningCollector
         implements WarningCollector
 {
     private final Set<PrestoWarning> warnings = new LinkedHashSet<>();
+    private final QueryId queryId;
+
+    public DedupingWarningCollector(QueryId queryId)
+    {
+        this.queryId = requireNonNull(queryId, "queryId is null");
+    }
 
     public void record(WarningCode warning, String message)
     {
-        warnings.add(new PrestoWarning(warning, message));
+        warnings.add(new PrestoWarning(
```

Comment:
This formatting looks weird. I think you can just put the entire statement on the same line.